### PR TITLE
Clarify org.gradle.util import deprecation

### PR DESCRIPTION
**Pull request by oehme**
Tell the user which build file is affected, so they can fix
the warning more easily.

### CODE REVIEW

**Comment by eskatos** [comment]
@oehme Kotlin DSL v0.12.0 is already integrated in `release` and it is not using `ImportsReader.importPackages`.

**Comment by oehme** [comment]
Got some failing tests, seems like we actually have different ordering on the "importedPackages" list and the "nameToFqn" map. I'll investigate tomorrow morning, probably means RC1 one day later.

File: subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/DefaultScriptCompilationHandler.java [hunk 1]
```diff
@@ -281,24 +279,24 @@ public URL loadGroovySource(String filename) throws MalformedURLException {
...
             this.resolveVisitor = resolveVisitor;
             this.progressCallback = new ProgressCallback() {
                 @Override
                 public void call(ProcessingUnit context, int phase) throws CompilationFailedException {
                     if (phase == Phases.CANONICALIZATION) {
                         Set<String> deprecatedImports = resolveVisitor.getDeprecatedImports();
                         if (!deprecatedImports.isEmpty()) {
-                            DeprecationLogger.nagUserWith("Using " + Joiner.on(",").join(deprecatedImports)
-                                + " from the private org.gradle.util package without an explicit import is deprecated. Please either stop using these private classes (recommended) or import them explicitly at the top of your build file (not recommended). The implicit import will be removed in Gradle 5.0");
+                            DeprecationLogger.nagUserWith(StringUtils.capitalize(script.getDisplayName()) + " is using " + Joiner.on(" and ").join(deprecatedImports)
```
Comment (blindpirate) [inline subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/DefaultScriptCompilationHandler.java]: Looks like we need one more test with other script name, i.e. `apply 'other.gradle'` and check default import in `other.gradle` .

File: subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GradleResolveVisitor.java [hunk 2]
```diff
@@ -572,13 +570,9 @@ private boolean resolveFromDefaultImports(ClassNode type, boolean testDefaultImp
...
                 }
             }
-            for (String gradlePublicPackage : gradlePublicPackages) {
-                if (resolveFromResolver(type, gradlePublicPackage + "." + name)) {
-                    if ("org.gradle.util".equals(gradlePublicPackage)) {
-                        deprecatedImports.add(name);
-                    }
-                    return true;
-                }
+            if (resolveFromResolver(type, "org.gradle.util" + "." + name)) {
```
Comment (blindpirate) [inline subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GradleResolveVisitor.java]: Looks like this changes the original behavior. `for (String gradlePublicPackage : gradlePublicPackages)` loop was trying to resolve a symbol from default packages. Without this, lots of default imports couldn't be resolved (e.g. `org.gradle.api.*`).

File: subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy [hunk 3]
```diff
@@ -22,6 +22,10 @@ import spock.lang.Issue
 @Issue('https://github.com/gradle/gradle/issues/1793')
 class GradleUtilDefaultImportDeprecationIntegrationTest extends AbstractIntegrationSpec {
 
+    def setup() {
+        executer.requireOwnGradleUserHomeDir()
```
Comment (blindpirate) [inline subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy]: Why is this required?

File: subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GradleResolveVisitor.java [hunk 4]
```diff
@@ -555,7 +553,7 @@ private boolean resolveFromDefaultImports(ClassNode type, boolean testDefaultImp
             String name = type.getName();
             List<String> fqn = simpleNameToFQN.get(type.getName());
             if (fqn != null) {
-                if (fqn.size() == 1 && resolveFromResolver(type, fqn.get(0))) {
```
Comment (blindpirate) [inline subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GradleResolveVisitor.java]: Why is this changed? If we need to change it, we'd better create a test to cover this scenario.

File: subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/DefaultScriptCompilationHandler.java (hunk 1 above)
Comment (oehme) [reply subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/DefaultScriptCompilationHandler.java]: Done

File: subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy [hunk 5]
```diff
@@ -57,7 +61,24 @@ task noop{
         succeeds('noop')
 
         then:
-        outputContains("Using GradleVersion from the private org.gradle.util package without an explicit import is deprecated. Please either stop using these private classes (recommended) or import them explicitly at the top of your build file (not recommended). The implicit import will be removed in Gradle 5.0")
+        outputContains("build.gradle' is using GradleVersion from the private org.gradle.util package without an explicit import. Please either stop using these internal classes (recommended) or import them explicitly at the top of your build file. The implicit import is deprecated and will be removed in Gradle 5.0")
```
Comment (blindpirate) [inline subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy]: Nit-pick: have you noticed the `"build.gradle'` is half-single-quote and half-double-quote?

File: subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy (hunk 3 above)
Comment (oehme) [reply subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy]: The test relies on an empty compiled script cache. If the script is cached, it won't be compiled and there will be no warning.

This is working on CI because we do clean builds, but it fails locally if you execute the test more than once.

File: subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy (hunk 5 above)
Comment (oehme) [reply subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy]: It's not, the double quote is for the String, the single quote is at the end of the file name. I didn't want to test for the full file name here.

File: subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy (hunk 3 above)
Comment (blindpirate) [reply subprojects/core/src/integTest/groovy/org/gradle/groovy/scripts/GradleUtilDefaultImportDeprecationIntegrationTest.groovy]: Good catch!
